nodejs/week3/practices/exercise-template/src/backend/api/concerts.js

```diff
@@ -0,0 +1,169 @@
+const express = require("express");
+const { request } = require("../app");
+const router = express.Router();
+const knex = require("../database");
+
+
+router.get("/", async (request, response) => {
+  const concerts = await knex("concerts");
+ 
+  try {
+    // knex syntax for selecting things. Look up the documentation for knex for further info
+          if(request.query.hasOwnProperty("id")){
+            
+            const requestId=Number(request.query.id);
+            const concertFinder = concerts.some(concert=> concert.id ===requestId);
+            const requestedConcert = await concerts.filter(concert=> concert.id ===requestId);
+    
+            if(concertFinder){
+              response.json(requestedConcert);    
+            }else {
+              response.status(404).send('no concert match your request');
+            }
+
+    
+          }else if(request.query.hasOwnProperty("maxPrice")){
+            
+            const requestPrice=Number(request.query.maxPrice);
+            const priceFinder = concerts.some(concert=> Number(concert.price) < requestPrice);
+            const requestedConcert = await concerts.filter(concert=> Number(concert.price)<requestPrice);
```

Comment:
same here with ``priceFinder`` and ``requestedConcert``. the latter is enough.
You're just repeating yourself, of course array method ``some`` brings out a different output (true or false) but the filter is enough.